### DISCUSSION TITLE
Extend authentication of the backend connection

### DISCUSSION THREAD

**Discussion by jfernandezsqs** [International-Data-Spaces-Association/DataspaceConnector#122] [Q&A] [answered]
"The DSC currently only supports Basic Auth as authentication mechanism for external data sources. To enable further connection, we need the authentication option via an API key transmitted via the http header accordingly."

Does the Dataspace connector support authentication option via an API key transmitted via the http header accordingly?

**Comment by juliapampus** [comment, marked as answer]
@jfernandezsqs It does, as stated [here](https://international-data-spaces-association.github.io/DataspaceConnector/CommunicationGuide/v6/Provider#option-2-add-remote-data), You can add API-key authentication. If you need any other type of authentication or want to support another backend than a REST API, you can use all features of Camel, as described [here](https://international-data-spaces-association.github.io/DataspaceConnector/CommunicationGuide/v6/Camel#create-a-route-for-fetching-data).

**Comment by jfernandezsqs** [comment]
I have followed the documentation detailed [here](https://international-data-spaces-association.github.io/DataspaceConnector/CommunicationGuide/v6/Camel#create-a-route-for-fetching-data) with [Step 2](https://international-data-spaces-association.github.io/DataspaceConnector/CommunicationGuide/v6/Camel#step-2-create-a-generic-endpoint) location `https://localhost:7080/database` that is my Dataspace Connector deployment H2 database for testing this requirement.
When I try [Step 5](https://international-data-spaces-association.github.io/DataspaceConnector/CommunicationGuide/v6/Camel#step-5-link-the-generic-endpoint-to-the-route) it is giving code 400 Invalid input. These are the DSC logs;
```
2022-03-16T13:42:17,692 [https-jsse-nio-7080-exec-6] WARN - Invalid input. [exception=(JSON parse error: Cannot deserialize value of type `java.net.URI` from Array value (token `JsonToken.START_ARRAY`); nested exception is com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize value of type `java.net.URI` from Array value (token `JsonToken.START_ARRAY`)\n at [Source: (PushbackInputStream); line: 1, column: 1])]
```
I have detailed the created route-uuid and in the request body introduced the generic endpoint’s self-link as specified in documentation.
Is there something wrong with the location backend-url?
